cluster/core/src/main/scala/com/lightbend/lagom/internal/cluster/JoinClusterImpl.scala

```diff
@@ -29,14 +32,37 @@ private[lagom] object JoinClusterImpl {
     if (cluster.settings.SeedNodes.isEmpty && joinSelf)
       cluster.join(cluster.selfAddress)
 
+    val SUCCESSFUL_EXIT = 0
+    val exitStatus = new AtomicInteger(SUCCESSFUL_EXIT)
+
+    val isExiting = new AtomicBoolean(false)
+
     if (exitJvm) {
```

Comment:
I think this whole `if` statement should be removed.

Considering this is Lagom code, I can think of two reasons why the actor system is terminating:

- the node is downed from the cluster, in which case the `Sys-exiting-from-akka-termination` should not be run, or
- the lagom process (aka, the play server) is being shutdown, in which case it's the `ApplicationLifecycle`'s responsibility to shutdown the JVM. That is, the `ApplicationLifecycle` must terminate the actor system but I don't the actor system should be responsible for exiting the JVM in the scope of a Lagom/Play process

Reply:
This makes sense to me, but my understanding of the overall lifecycle is pretty incomplete. @jroper @patriknw do you have any thoughts about this?

Reply:
I think you should spend more time on it when you have updated to Akka 2.5 and the coordinated shutdown. This will be obsolete.